### PR TITLE
Add llvm support for pulpissimo

### PR DESCRIPTION
**Pull request by zhouxinyu0723**

### CODE REVIEW

**Review by PhilippvK — APPROVED** [review]

File: cmake/toolchains/gcc_gvsoc_pulpissimo.cmake [hunk 1]
```diff
@@ -10,11 +10,38 @@ SET(CMAKE_LINKER ${TC_PREFIX}ld${EXE_EXT})
...
+SET(CV_LDFLAGS "\
+-nostartfiles \
+-Wl,--gc-sections \
+-Wl,-Map,memory.map") # -Wl,--print-gc-sections
+
+# LIST(APPEND CV_LDFLAGS " -Wstack-usage=1024")
+
+# the following is transfered from PULP-FreeRTOS repository
+# https://github.com/pulp-platform/pulp-freertos/blob/master/default_targets.mk#L77-L87
+SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${CV_CFLAGS} ${CV_CPPFLAGS} -DSTDIO_FAKE=2 -DSTDIO_UART=1 -DSTDIO_NULL=0 -DCONFIG_STDIO=2 -Og ")
```
Comment (PhilippvK) [inline cmake/toolchains/gcc_gvsoc_pulpissimo.cmake:44]: @zhouxinyu0723 What are those `-Og` for at the end of each line?

Comment (zhouxinyu0723) [reply cmake/toolchains/gcc_gvsoc_pulpissimo.cmake]: It means optimizing code for debugging. This should be removed and I have pushed a new commit removing them.

**Comment by zhouxinyu0723** [comment]
Should I also apply the changes to the target pulp. The current changes are for pulpissimo

**Comment by PhilippvK** [comment]
> Should I also apply the changes to the target pulp. The current changes are for pulpissimo

I would like to test the pulpissimo patch first.

**Comment by PhilippvK** [comment]
@zhouxinyu0723 I pushed a few updates to you branch. The Pulpissimo support seems to work like expected. Feel free to create the `pulp` version of the toolchain now as well.

**Comment by zhouxinyu0723** [comment]
A question, should the "-fuse-ld=lld" be added to the link flags?

**Comment by PhilippvK** [comment]
> A question, should the "-fuse-ld=lld" be added to the link flags?

@zhouxinyu0723  Yes!

**Comment by PhilippvK** [comment]
Will you add the pulp patch here or to a new pr?

**Comment by zhouxinyu0723** [comment]
> Will you add the pulp patch here or to a new pr?

I would like to add it here because they are very similar to each other.

File: cmake/targets/gvsoc/PulpTarget.cmake [hunk 2]
```diff
@@ -36,6 +36,7 @@ MACRO(ADD_EXECUTABLE_GVSOC_PULP_INTERNAL TARGET_NAME ADD_PLATFORM_FILES)
             -T ${GVSOC_TARGET_LIB_DIR}/link.ld \
             "
         )
+        SET(ARA_MACRO_ALREADY_EXECUTED ON)
```
Comment (PhilippvK) [inline cmake/targets/gvsoc/PulpTarget.cmake:39]: Previous lines still mention `GVSOC_PULP_MACRO_ALREADY_EXECUTED ` is that correct?

**Comment by PhilippvK** [comment]
@zhouxinyu0723  ready to merge or is llvm support for PULP missing?

**Comment by zhouxinyu0723** [comment]
> 

LLVM support for pulp is also within this PR. Basically, `cmake/toolchains/llvm_gvsoc_pulp.cmake` is copied from `cmake/toolchains/llvm_gvsoc_pulpissimo.cmake`. The only difference between pulp and pulpissmio in mlonmcu-sw repo is the BSP packages they are using

**Pull request merged**